hive.go

```diff
@@ -86,6 +86,18 @@ func mainInHost(daemon *docker.Client, overrides []string, cacher *buildCacher)
 	// Retrieve the versions of all clients being tested
 	if results.Clients, err = fetchClientVersions(daemon, *clientPattern, cacher); err != nil {
 		log15.Crit("failed to retrieve client versions", "error", err)
+		b, ok := err.(*buildError)
+		if ok {
+			results.Clients = make(map[string]map[string]string)
+			results.Clients[b.Client()] = make(map[string]string)
+			results.Clients[b.Client()] = map[string]string{"Failure": "Error building docker container for client"}
+		}
+		out, errMarshal := json.MarshalIndent(results, "", "  ")
+		if errMarshal != nil {
+			log15.Crit("failed to report results. Docker Failed build.", "error", err)
+			return err
+		}
+		fmt.Println(string(out))
```

Comment:
Shouldn't the entire output be within the same `if ok{` - clause? As it is, you'll almost certainly be marshalling a `nil` object if some other type of error occurred.

Reply:
I think you're right. I'll move this code so it's under the same `if ok{` clause.